File: DAC_API/Models/DTO/Submission/CreateSubmissionDTO.cs
using System.ComponentModel.DataAnnotations;

namespace DAC_API.Models.DTO.Submission {
    public class CreateSubmissionDTO {
        [Required(ErrorMessage = "User ID is required")]
        public int UserId { get; set; }

        [Required(ErrorMessage = "Challenge ID is required")]
        public int ChallengeId { get; set; }

        [Required(ErrorMessage = "Photo URL is required")]
        [Url(ErrorMessage = "Invalid URL format")]
        public string PhotoUrl { get; set; }
    }
}

File: DailyAdventureChallenge/app/src/main/java/com/example/dailyadventurechallenge/data/dto/Submission/CreateSubmissionDTO.kt
package com.example.dailyadventurechallenge.data.dto.Submission

import com.google.gson.annotations.SerializedName

data class CreateSubmissionDTO(
    @SerializedName("userId")
    val userId: Int,
    @SerializedName("challengeId")
    val challengeId: Int,
    @SerializedName("photoUrl")
    val photoUrl: String
)
